src/darwin/Framework/CHIP/MTRThreadOperationalDataset.mm

```diff
@@ -38,15 +39,15 @@ - (nullable instancetype)initWithNetworkName:(NSString *)networkName
                                extendedPANID:(NSData *)extendedPANID
                                    masterKey:(NSData *)masterKey
                                         PSKc:(NSData *)PSKc
-                                     channel:(uint16_t)channel
+                               channelNumber:(NSNumber *)channelNumber
                                        panID:(NSData *)panID
 {
     if (self = [super init]) {
         _networkName = networkName;
```

Comment:
Should be calling `copy` on all these parameters if we want to be tidy about these things (`_networkName` down to `_panID`)

Reply:
Yep, will do.